src/components/MemoryGraph.js

```diff
@@ -0,0 +1,106 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { Line } from 'react-chartjs-2';
+import useInterval from '../hooks/use-interval';
+
+/* this line chart graphs the last 10 data points we've seen,
+and updates the lower and upper bound of the range to be equal
+to the min and max of this dataset, respectively. */
+const numData = 10;
+
+const MemoryGraph = ({ next }) => {
+  const [numArray, addToNumArray] = useState([]);
+  const [min, setMin] = useState(0);
+  const [max, setMax] = useState(2500000);
+
+  const setRange = (newArr) => {
+    setMin(Math.min(...newArr));
+    setMax(Math.max(...newArr));
+  };
+
+  useInterval(() => {
+    // const answer = (parseFloat(next) / total) * 100
```

Comment:
do we need to use useInterval here? the parent element is using useCpu which uses useInterval, and you are getting state from the parent. i'm not sure you need useInterval

Reply:
am now using useEffect